File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        Arrays.stream(nums).sorted().forEach(num -> {
            if (map.containsKey(num - 1)) {
                int val = map.get(num - 1) + 1;
                map.put(num, val);
            } else {
                map.put(num, 1);
            }
        });

       return map.values()
               .stream()
               .max(Integer::compareTo)
               .orElse(0);
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        for i, p1 in enumerate(prices):
            p2 = sorted(prices[i:])[-1]
            if p1 < p2:
                max_profit = max(max_profit, p2 - p1)
            

        return max_profit
